CMakeLists.txt

```diff
@@ -89,25 +89,18 @@ if(NOT MSVC)
 endif()
 
 if(Boost_FOUND)
-  get_property(SOCI_COMPILE_DEFINITIONS
-    DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    PROPERTY COMPILE_DEFINITIONS)
-
-  list(APPEND SOCI_COMPILE_DEFINITIONS "HAVE_BOOST=1")
-  list(APPEND SOCI_COMPILE_DEFINITIONS "BOOST_ALL_NO_LIB")
```

Comment:
On windows I also got an error on link but when calling cmake if I set `Boost_USE_STATIC_LIBS=ON` everything is fine.